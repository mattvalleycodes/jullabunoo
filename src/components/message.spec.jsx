import * as React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import Message from "./message";

describe("<Message /> component", () => {
  test("by default, renders in info mode", () => {
    const message = shallow(<Message title="Hello, world!" />);

    expect(message.hasClass("message--info")).toEqual(true);
  });

  test("renders error messages in error mode", () => {
    const message = shallow(<Message type="error" title="Hello, world!" />);

    expect(message.hasClass("message--error")).toEqual(true);
  });

  test("renders the message information", () => {
    const message = renderer
      .create(<Message type="error" title="Hello, World!" />)
      .toJSON();

    expect(message).toMatchSnapshot();
  });
});
