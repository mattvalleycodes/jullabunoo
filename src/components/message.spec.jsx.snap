// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> component renders the message information 1`] = `
<div
  className="message message--error"
>
  Hello, World!
</div>
`;
